### DISCUSSION TITLE
How to choose way in dependence of existing element

### DISCUSSION THREAD

**Discussion by Rutol** [cypress-io/cypress#17030] [Questions and Help]
There are web interface, where most of elements could be identified only by text, i.e. using cy.contains().

In one case I have element with specific test (for example "PolterGates") and in other case haven't such element.

I need realise construction:

if (element with text "PolterGates" exist) {
set of action number 1
} else {
set of action number 2
}

How to realize it?
should() not suite, because in case of negative assert test fails.

PLEASE DON'T SEND link to these page in documentation 
https://docs.cypress.io/guides/core-concepts/conditional-testing
because I have read it several times and can't find appropriate solution.

**Comment by bahmutov** [comment]
@Rutol would this be helpful instead https://glebbahmutov.com/cypress-examples/7.5.0/recipes/conditional-testing.html ?

**Reply by Rutol** [reply]
Коллега, не совсем! 
Возможно я не понял Ваш js-код, но ситуация такая.
Срабатывает поиск, в случае если результатов нет, пишет соответствующий текст.
Практически все объекты на странице не имеют уникальных имен и имеют динамические id (да, я тоже могу сказать про веб-программера много "хорошего", но ситуацию это не поменяет)
и фактически надо сделать такую штуку:
if(объект с текстом "никого нет дома" найден) {
набор действий 1
} else {
набор действий 2 }

**Comment by bahmutov** [comment]
How do you know when the search has finished?
